packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -253,84 +254,123 @@ export const addPositionPropertiesToSections = <TSection extends FieldSection>(
+      return localeText.fieldSecondsPlaceholder();
+    }
+
+    case 'am-pm': {
+      return localeText.fieldMeridiemPlaceholder();
+    }
 
-  return utils.formatByString(date, format);
+    default: {
+      return currentTokenValue;
+    }
+  }
 };
 
 export const splitFormatIntoSections = <TDate>(
   utils: MuiPickerFieldAdapter<TDate>,
+  localeText: PickersLocaleText<TDate>,
   format: string,
   date: TDate | null,
 ) => {
   let currentTokenValue = '';
   const sections: Omit<FieldSection, 'start' | 'end'>[] = [];
   const expandedFormat = utils.expandFormat(format);
 
+  const commitCurrentToken = () => {
+    if (currentTokenValue === '') {
+      return;
+    }
+
+    const sectionConfig = getDateSectionConfigFromFormatToken(utils, currentTokenValue);
+    const sectionValue = date == null ? '' : utils.formatByString(date, currentTokenValue);
+
+    sections.push({
+      ...sectionConfig,
+      formatValue: currentTokenValue,
+      value: sectionValue,
+      placeholder: getSectionPlaceholder(utils, localeText, sectionConfig, currentTokenValue),
+      separator: '',
+      edited: false,
+    });
+
+    currentTokenValue = '';
+  };
+
   for (let i = 0; i < expandedFormat.length; i += 1) {
     const char = expandedFormat[i];
-    if (!char.match(/([A-zÀ-ú]+)/g)) {
-      if (currentTokenValue === '') {
-        sections[sections.length - 1].separator += char;
-      } else {
-        const dateForCurrentToken = formatDateWithPlaceholder(utils, date, currentTokenValue);
-        if (dateForCurrentToken === currentTokenValue) {
-          sections[sections.length - 1].separator += currentTokenValue;
-          currentTokenValue = '';
-        } else {
-          const sectionConfig = getDateSectionConfigFromFormatToken(utils, currentTokenValue);
-
-          sections.push({
-            ...sectionConfig,
-            formatValue: currentTokenValue,
-            value: dateForCurrentToken,
-            emptyValue: sectionConfig.dateSectionName,
-            separator: char,
-            edited: false,
-          });
-          currentTokenValue = '';
-        }
-      }
-    } else {
-      currentTokenValue += char;
-    }
 
-    if (i === expandedFormat.length - 1) {
-      const dateForCurrentToken = formatDateWithPlaceholder(utils, date, currentTokenValue);
-      if (dateForCurrentToken === currentTokenValue) {
-        sections[sections.length - 1].separator += currentTokenValue;
-      } else {
-        const sectionConfig = getDateSectionConfigFromFormatToken(utils, currentTokenValue);
-
-        sections.push({
-          ...sectionConfig,
-          formatValue: currentTokenValue,
-          value: dateForCurrentToken,
-          emptyValue: sectionConfig.dateSectionName,
-          separator: null,
-          edited: false,
-        });
+    if (char.match(/([A-zÀ-ú]+)/g)) {
+      currentTokenValue += char;
+    } else {
+      if (currentTokenValue !== '') {
+        commitCurrentToken();
       }
```

Comment:
You already do this verification in the function
```suggestion
      commitCurrentToken();
```